### ISSUE TITLE
[Bug]:MiMo Code mimo_v2_omni.py ERROR

### ISSUE THREAD

**Issue by flower-with-safe** [vllm-project/vllm#47864]
### Your current environment

<details>
<summary>error code  <code> _forward_window_attn </code></summary>
def _forward_window_attn(self, q, k, v, cu_seqlens, max_seqlen):
    """Window attention via flash_attn_varlen_func with window_size."""
    from vllm.vllm_flash_attn import flash_attn_varlen_func
 
    w = self.visual_token_window_size
    output = flash_attn_varlen_func(
        q, k, v,
        cu_seqlens_q=cu_seqlens,
        cu_seqlens_k=cu_seqlens,
        max_seqlen_q=max_seqlen,
        max_seqlen_k=max_seqlen,
        softmax_scale=self.scale,
        causal=False,
        window_size=[w, w],
    )
    return output
</details>

<details>
<summary>correct code  <code> _forward_window_attn </code></summary>
def _forward_window_attn(self, q, k, v, cu_seqlens, max_seqlen):
    """Window attention via flash_attn_varlen_func with window_size and sinks."""
    from vllm.vllm_flash_attn import flash_attn_varlen_func
 
    w = self.visual_token_window_size
    s_aux = None
    if self.sinks is not None:
        head_start = self.tp_rank * self.num_heads_per_partition
        head_end = head_start + self.num_heads_per_partition
        s_aux = self.sinks[head_start:head_end]
    output = flash_attn_varlen_func(
        q, k, v,
        cu_seqlens_q=cu_seqlens,
        cu_seqlens_k=cu_seqlens,
        max_seqlen_q=max_seqlen,
        max_seqlen_k=max_seqlen,
        softmax_scale=self.scale,
        causal=False,
        window_size=[w, w],
        fa_version=3,
        s_aux=s_aux,
    )
    return output
</details>

### 🐛 Describe the bug

mimo_v2_omni.py code error

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by almogtavor** [comment]
@flower-with-safe opened #49815 for this.

You are right that the window path drops the sinks. `self.sinks` is only allocated when the block is not in `fullatt_block_indexes`, so it exists exactly for the blocks that call `_forward_window_attn`, and that function never read it. With the released vision config that is 24 of 28 blocks.

One change from your patch though. Passing the sinks as `s_aux` is a different operation. `s_aux` adds a null logit to the softmax denominator, while the reference adds `sinks[h]` to the logit of key 0. Measured on an H100 in bf16 against the reference:

```
current, sinks dropped     rel error 4.8e-02
s_aux patch                rel error 5.2e-02
this PR                    rel error 2.2e-03
```

So `s_aux` comes out slightly worse than doing nothing. 2.2e-03 is just the bf16 noise floor.

flash attention cannot express a bias on a single key. My first attempt recomputed the affected rows with SDPA, but that mixes two attention implementations in one call, so the PR instead corrects the output in closed form using the softmax LSE that flash attention already returns. Biasing one key only rescales that key's softmax term, so `out' = (out + p0 * (exp(s) - 1) * v0) / (1 + p0 * (exp(s) - 1))` with `p0 = exp(logit0 - lse)`. Only queries within the window of key 0 are affected. It is not completely free, the key 0 logit gets recomputed outside the kernel, and adding real bias support to the flash attention wrapper would be cleaner if maintainers want to go that way.

Checked over lengths 128 to 4096, multi sequence batches, GQA and TP=2 on both ranks.

The omni checkpoint is not public so I compared against the vision attention in `modeling_mimo_v2.py` from `XiaomiMiMo/MiMo-V2.5`, which has the same vision config surface. If you have the omni weights an end to end check would be useful.
